src/internal/m365/collection/groups/export.go

```diff
@@ -198,3 +213,146 @@ func makeMinimumChannelMesasge(item models.ChatMessageable) minimumChannelMessag
 		Subject:              ptr.Val(item.GetSubject()),
 	}
 }
+
+//-------------------------------------------------------------
+// Conversation Posts
+//-------------------------------------------------------------
+
+// streamConversationPosts adds the post items into the export stream channel.
+func streamConversationPosts(
+	ctx context.Context,
+	drc []data.RestoreCollection,
+	backupVersion int,
+	cec control.ExportConfig,
+	ch chan<- export.Item,
+	stats *metrics.ExportStats,
+) {
+	defer close(ch)
+
+	errs := fault.New(false)
+
+	for _, rc := range drc {
+		ictx := clues.Add(ctx, "path_short_ref", rc.FullPath().ShortRef())
+
+		for item := range rc.Items(ctx, errs) {
+			itemCtx := clues.Add(ictx, "stream_item_id", item.ID())
+
+			// Trim .data suffix from itemID. Also, we don't expect .meta files
+			// here since details are not persisted for metadata files.
+			trimmedID := strings.TrimSuffix(item.ID(), metadata.DataFileSuffix)
+			exportName := trimmedID + ".eml"
+
+			postMetadata, err := fetchAndReadMetadata(itemCtx, trimmedID, rc)
+			if err != nil {
+				ch <- export.Item{
+					ID:    item.ID(),
+					Error: err,
+				}
+
+				continue
+			}
+
+			reader := item.ToReader()
+			content, err := io.ReadAll(reader)
+
+			reader.Close()
+
+			if err != nil {
+				ch <- export.Item{
+					ID:    item.ID(),
+					Error: err,
+				}
+
+				continue
+			}
+
+			// Convert JSON to eml.
+			email, err := eml.FromJSONPostToEML(itemCtx, content, postMetadata)
+			if err != nil {
+				err = clues.Wrap(err, "converting JSON to eml")
+
+				logger.CtxErr(itemCtx, err).Info("processing collection item")
+
+				ch <- export.Item{
+					ID:    item.ID(),
+					Error: err,
+				}
+
+				continue
+			}
+
+			emlReader := io.NopCloser(bytes.NewReader([]byte(email)))
+
+			stats.UpdateResourceCount(path.ConversationPostsCategory)
+			body := metrics.ReaderWithStats(emlReader, path.ConversationPostsCategory, stats)
+
+			ch <- export.Item{
+				ID:   item.ID(),
+				Name: exportName,
+				Body: body,
+			}
+		}
+
+		items, recovered := errs.ItemsAndRecovered()
+
+		// Return all the items that we failed to source from the persistence layer
+		for _, item := range items {
+			ch <- export.Item{
+				ID:    item.ID,
+				Error: &item,
+			}
+		}
+
+		for _, err := range recovered {
+			ch <- export.Item{
+				Error: err,
+			}
+		}
+	}
+}
+
+func fetchAndReadMetadata(
+	ctx context.Context,
+	itemID string,
+	fin data.FetchItemByNamer,
+) (groupMeta.ConversationPostMetadata, error) {
+	metaName := itemID + metadata.MetaFileSuffix
+
+	ictx := clues.Add(ctx, "meta_file_name", metaName)
```

Comment:
`ictx` isn't a necessary cast unless we're inside a loop.  All other clues additions can replace the `ctx`.
```suggestion
	ctx = clues.Add(ctx, "meta_file_name", metaName)
```